### DISCUSSION TITLE
Call for help: Let's build some quick ass sentences for the new `HassSetPosition` intent

### DISCUSSION THREAD

**Discussion by jlpouffier** [OHF-Voice/intents#2055] [_General]
Yo @home-assistant/language-leaders 👋🏻

Normally, we all copy the idea of the English sentences to create our own sentences in our own language, English proving guidance most of the time.
For the `HassSetPosition`, English was a bit late to get all the use cases created for this release and I saw a few languages replicating the **ONE** sentence that exists in English.

I would recommend you to check what has been done in French, I can translate roughly what I did here to inspire you.
I opened this discussion because while writing the release note for. 2024.3, I noticed that this intent is quite week for now.

First I split the file `homeassistant_HassSetPosition.yaml` into `cover_HassSetPosition.yaml` and `valve_HassSetPosition.yaml` to get more granular control on what I was doing.

On the valve side, I decided to keep it extremely simple and only allow users to set a valve position by targeting **ONE** entity by its name. 
Like `set the irrigation valve position to 50%` and that's it.
This file is here and it's pretty simple, one use-case: Name targeting: https://github.com/home-assistant/intents/blob/main/sentences/fr/valve_HassSetPosition.yaml
_Why: valve as a domain is too new to understand exactly what people want to do with it._

On the cover side, we did:
- Targeting a cover by name: `set the blue curtain position to 50%`
- Targeting a cover by name + area: `set the living room blue curtain position to 50%`
- Targeting covers by device class + area: `set (all) the living room shutterS position to 50%`
- targeting covers by device class for the whole house: `set (all) the shadeS position to 50%`
- Context area awareness but device class: `set  the curtainS position (in here) to 50%`

Even if you cannot read French, the comments show the structuring of the file here:
https://github.com/home-assistant/intents/blob/main/sentences/fr/cover_HassSetPosition.yaml

Note (This one is more on a per language basis), I am far from being an expert in your language: 
In French we decided to use interchangeably the verbs `set`, `open`, and  `close` for setting a cover position 
- `set the blue curtain position to 50%`
- `open the blue curtain to 80%`
- `close the blue curtain to 20%`

We assumed that if users want to set a position that is smaller than the current position, they may use a "closing" verb, and vice versa.

I remain available if you need help!

**Comment by Uriziel01** [comment]
Yo JLo. So in the end `open the curtains to 10%` would have the exact same effect as `close the curtains to 10%`? Seems quite confusing, maybe it's just something wrong with Polish language in general but that's my perspective 🙂

**Reply by tetele** [reply]
They would have the same effect, indeed. However, I doubt someone would have their curtains completely shut and say "close the curtain to 10%" and be amazed that they opened a bit :P

**Reply by jlpouffier** [reply]
In French yes that is what we did - It's aligned with Alexa for example.

Our reasoning was that you won't use these two sentences in the same context.

If your curtains are fully open (100%), then you may say: Close the curtains to 10%
If they are fully closed (0%), then you may say: Open the curtains to 10%
(Even if technically they do the same thing: Set potion to 10%)

Check Alexa, or Siri, Google Assistant, etc in your own language.
But again: I am in no way an expert in every language, so this is clearly up to you. ;)

**Reply by tetele** [reply]
However, you're the language leader, @Uriziel01, so you know best what fits Polish and what doesn't. If it doesn't make sense in your language, don't use the opening/closing verbs

**Reply by Uriziel01** [reply]
Even as a language leader I can't speak for all the people of Poland 😂 Will ask some more people on discord what is their take on this, thanks for all the insights @jlpouffier @tetele 🙂

**Reply by dontinelli** [reply]
I agree with @jlpouffier. It would be too complicated (and not yet possible from a technical point of view) to distinct the wording depending on the starting position. And for me open/close are more natural than set. But this might be different for other persons and/or in other languages.

**Reply by salleq** [reply]
If the user doesn't know 100% is fully open and 0% is fully closed and thinks it the other way? "Close my curtain 100%" -> curtain opens  completely.

**Reply by jlpouffier** [reply]
I just did @tetele tests on Alexa just to verify :) 

Shutters fully open 100%
- Close the shutters to 80%: The shutters close to 80% and it's coherent ✅
- Open the shutters to 90%: The shutters open to 90% and it's coherent ✅
- Open the shutters to 80%: The shutters close to 80% and it's not coherent ❌
- Close the shutters to 90%: The shutters open to 90% and it's not coherent ❌

Basically: They don't care, it's up to the user to pick what's natural

Funny story, a long time ago I was developing intents for Snips (A long-gone project) and I did something that I felt was more coherent.

When I was saying "OPen to X%", I was setting the position to X
When I was saying "Closee to X", I was setting the position to 100-X
It was a disaster, I kept forgetting which on was which. Awefull UX

**Reply by cibernox** [reply]
FWIW, in spanish I often say "Open the shutters to 50%" to open them. 

Maybe my shutters were already at 60%, and that results on the shutters actually closing a little bit more. Or maybe there are several shutters each one opened a different percentage and some of them open more while others close a bit, but that's fine, the "spirit" of the intent is that I want my shutter or shutters at 50%, regardless of how open they are.

Also, at least for my rolling shutters, it's actually hard to estimate how open they are because being "visually" at 50% is not exactly the same as being 50%, because there's some amount of play with the slates of the shutter that makes the first 10% not actually lift the shutter but widen the gap between the slates.

In short, I think we should allow for open/close/set as valid verbs for controlling the position of covers. If the user specifies a percentaje, it's clear what their intent is, regardless of what verb they used and how opened the cover was.

**Reply by jlpouffier** [reply]
Remark: (Even if I like this discussion) My post was merely a call for action on the fact we're missing, today, a bunch of sentences in a bunch of languages that are supported ;)

**Comment by andrejs2** [comment]
Agree with interchangeability of `set`, `open `and `close`, the effect is the same as per @Uriziel01 , but user might be tempted to use `close the curtain to 10 %` when curain is opened and vice versa.

**Comment by mib1185** [comment]
What is about "set|open|close" `by` sentences (_make it relativ to current position_)?
Example: Blind is opened 80%
- `close blind by 20 %` should set it to 60%
- `open blind by 10 %` should set it to 90%

or should this go into another discussion (_same logic should count for valves and volume_)? 🙂

**Reply by andrejs2** [reply]
I was just thinking about this last week when I translated the sentences :) I think that's an upgrade :)

**Reply by cibernox** [reply]
that is actually a good sentence I never thought off. Is that even possible? Can we do maths with the current percentage of openness?

**Reply by mib1185** [reply]
i guess this needs some pre-work in HA core as well as the intent core code itself 🤔

**Reply by tetele** [reply]
We need a service to incrementally alter a numeric state, which we don't have yet. It would be the same for temperatures (`increase the temperature by 1 degree`), volume (`turn down the volume by 5%`) etc.

**Reply by andrejs2** [reply]
The most useful in my opinion would this be for volume. I tell my kinds to put volume down for a notch and my 6-er understands :)

Mainly because I don't have fixed volume output because of TV and STB reciever; both can regulate volume and 5% on reciever output is not the same -dB as if TV is on 5% or 25%. Anyways...I strongly support your proposal  for volume at least.

**Reply by cibernox** [reply]
This should be proposed in https://github.com/home-assistant/architecture/discussions/
I think that setting the position is cool, but sometimes increasing or decreasing a position by some amount without actually knowing what its value is can be very useful, particularly with voice control.

**Reply by jlpouffier** [reply]
Remark: (Even if I like this discussion) My post was merely a call for action on the fact we're missing, today, a bunch of sentences in a bunch of languages that are supported ;)

**Reply by mib1185** [reply]
- arch discussion started -> https://github.com/home-assistant/architecture/discussions/1057

**Comment by v1k70rk4** [comment]
@jlpouffier, I'm already watching all of your PRs :) this is already a living thing in Hungarian. Thank you here as well.
